ipsource.go

```diff
@@ -238,6 +241,105 @@ func (u UPnP) GetIPs(ctx context.Context, _ IPVersions) ([]net.IP, error) {
 	return []net.IP{ip}, nil
 }
 
+// Command is an IP source that looks up the public IP addresses by
+// executing a script or command from your filesystem.
+//
+// The command must return the IP addresses comma spreaded in plain text.
+type Command struct {
+	// The command to execute.
+	Cmd string `json:"command,omitempty"`
+
+	// Arguments to the command. Placeholders are expanded
+	// in arguments, so use caution to not introduce any
+	// security vulnerabilities with the command.
+	Args []string `json:"args,omitempty"`
+
+	// The directory in which to run the command.
+	Dir string `json:"dir,omitempty"`
+
+	// How long to wait for the command to terminate
+	// before forcefully closing it. Default: 30s
+	Timeout caddy.Duration `json:"timeout,omitempty"`
+
+	logger *zap.Logger
+}
+
+// CaddyModule returns the Caddy module information.
+func (Command) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "dynamic_dns.ip_sources.command",
+		New: func() caddy.Module { return new(Command) },
+	}
+}
+
+func (c *Command) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	var (
+		unused string
+		cmd    string
+	)
+	if !d.AllArgs(&unused, &cmd) {
+		return d.ArgErr()
+	}
+	c.Cmd = cmd
+	return nil
+}
```

Comment:
See https://github.com/mholt/caddy-events-exec/blob/a5d053d9cd44b956f36a6ccfd3c0e0adf16cc7f1/module.go#L131

Reply:
What about `Dir`and `Timeout` aren't they parsed in this example?

Reply:
That plugin actually doesn't support `dir` and `timeout` via Caddyfile yet. That would typically be done with a "block" (i.e. `d.NextBlock(0)` and such). See `caddyfile.go` in this repo for examples of parsing a block.